### ISSUE TITLE
Using docker and passing a truststore/keystore in the container

### ISSUE THREAD

**Issue by Tzaphkiel** [edenhill/kafkacat#253]
I would like to use a **locally stored keystore/truststore** and pass it to the docker container so I can then run kafkacat with the enterprise SSL CA chain for example.

```sh
docker run --rm --network=host edenhill/kafkacat:1.5.0 kafkacat \
  -b "mybrokers:ports, ..." \
  -X security.protocol=SSL \
  -v -L -d broker,metadata,topic
```

like it is possible for kafdrop: https://github.com/obsidiandynamics/kafdrop#using-docker

Is this possible? I've searched a while the doc and issues but could not find it...

**Comment by edenhill** [comment]
Truststores and Keystores are Java concepts, while kafkacat uses OpenSSL which expects PEM files, etc.

Use keytool to extract certs from your trust and keystores and pass them as `-X ssl.certificate.location=..` etc.
https://github.com/edenhill/librdkafka/wiki/Using-SSL-with-librdkafka

**Comment by Tzaphkiel** [comment]
You are right about JAVA, My mistake. I'm however confused by the passing of a certificate to kafkacat, does that incluse own CAs' certificates as well or just client certificate used for authentication to the brokers?

**Comment by edenhill** [comment]
See  https://github.com/edenhill/librdkafka/blob/master/INTRODUCTION.md#ssl

ssl.ca.location is used by the client to verify the broker's certificate
ssl.certificate.location is the client's public key
ssl.key.location is the client's private key

None or not all of these needs to be used, based on your broker config.

Den tors 2 juli 2020 kl 12:35 skrev Sébastien <notifications@github.com>:

**Comment by dgersh** [comment]
Your links to the librdkafka for SSL would be a nice addition to the main README.md since I also made the same mistake trying to use my truststore information to connect via kafkacat.
